src/App.tsx

```diff
@@ -10,33 +11,34 @@ export const tabs = [
 ];
 
 export const App: React.FC = () => {
+  const [selectedTabId, setSelectedTab] = useState('tab-1');
+
+  useEffect(() => {
+    if (!tabs.map(tab => tab.id).includes(selectedTabId)) {
```

Comment:
The use of 'useEffect' here may lead to unnecessary rerenders. Consider moving the check inside the 'onTabSelected' function to only verify when a new tab is selected.

Think you don't need here this useEffect hook.